cloud/src/autogluon/cloud/predictor/cloud_predictor.py

```diff
@@ -0,0 +1,861 @@
+from ..utils.utils import rename_file_with_uuid
+from ..utils.constants import SAGEMAKER_TRUST_REPLATIONSHIP, SAGEMAKER_POLICIES, VALID_ACCEPT
+
+logger = logging.getLogger(__name__)
+
+cloud_path = Path(__file__).parent.parent.absolute()
+script_path = os.path.join(cloud_path, 'scripts')
+train_script = os.path.join(script_path, 'train.py')
+tabular_serve_script = os.path.join(script_path, 'tabular_serve.py')
+text_serve_script = os.path.join(script_path, 'text_serve.py')
+
+
+class CloudPredictor:
+
+    predictor_file_name = 'CloudPredictor.pkl'
+
+    def __init__(
+        self,
+        predictor_type,
+        role_arn=None,
+        local_output_path=None,
+        cloud_output_path=None,
+        verbosity=3
+    ):
+        """
+        Parameters
+        ----------
+        predictor_type: str
+            Type of predictor to be trained through SageMaker
+        role_arn: str
+            The role_arn you want to use to grant cloud predictor necessary permission. 
+            This role must have permission on AmazonS3FullAccess and AmazonSageMakerFullAccess.
+            If None, CloudPredictor will create one.
+        local_output_path: str
+            Path to directory where trained models and intermediate outputs should be saved
+            If None, CloudPredictor will create one.
+        cloud_output_path: str
+            Path to s3 location where trained models and intermediate outputs should be should be saved
+            If None, CloudPredictor will create one.
+        verbosity: int
+            Verbosity levels range from 0 to 4 to control how much information is displayed
+        """
+        assert predictor_type in ['tabular', 'text']
+        self.verbosity = verbosity
+        if self.verbosity is not None:
+            set_logger_verbosity(self.verbosity)
+        self.role = role_arn
+        if not self.role:
+            self.role = create_sagemaker_role_and_attach_policies(
+                role_name='ag_cloud_predictor_role',
+                trust_relationship=SAGEMAKER_TRUST_REPLATIONSHIP,
+                policies=SAGEMAKER_POLICIES
+            )
+        self.sagemaker_session = sagemaker.session.Session()
+        self.region = self.sagemaker_session.boto_region_name
+        self.path = self._setup_local_outputpath(local_output_path)
+        self.cloud_output_path = self._setup_cloud_outputpath(cloud_output_path)
+        self.predictor_type = predictor_type
+        self.endpoint = None
+        self.cloud_predictor_path = None
+        self.recent_predict_results_path = None
+        self.transformer = None
+        self.fit_job_name = None
+        self.transform_job_name = None
+
+    @property
+    def endpoint_name(self):
+        """
+        Return the CloudPredictor deployed endpoint name
+        """
+        if self.endpoint:
+            return self.endpoint.endpoint_name
+        return None
+
+    @property
+    def info(self):
```

Comment:
why is this a property? It isn't a property in TabularPredictor. Would recommend not having it be a property to enable more flexibility in future.